src/services/Utils.js

```diff
@@ -13,15 +13,15 @@ const RegisterCustomExtensions = function (AutodeskViewing, customExtensions) {
         let name = extensionNames[i];
 
         // If extension already registered
-        if(AutodeskViewing.theExtensionManager.getExtension(name) != null){
+        if (AutodeskViewing.theExtensionManager.isAvailable(name)) {
```

Comment:
wondering if this could be done in a way it supports v6 as well? Tbh don't think someone starting a new project will use a older version, but just in case?

Reply:
I guess it could look like something like that:

Where 
- 'window.LMV_VIEWER_VERSION' is a global variable containing the viewer's version number
- 'isExtensionAlreadyRegistered' is the function name responsible to tell if the extension is registered or not

```javascript
/**
 *
 * @param {Object} autodeskViewing Autodesk.Viewing
 * @param {Object} customExtensions Custom extensions
 */
const RegisterCustomExtensions = function (AutodeskViewing, customExtensions) {

    let extensionNames = Object.keys(customExtensions);

    let registeredExtensions = [];

    for (let i = 0; i < extensionNames.length; i++) {
        let name = extensionNames[i];

        // Get the function name that checks if the extension is already registered, depending on the viewer version
        const isExtensionAlreadyRegistered = /^7.[0-9]{1,}.[0-9]{1,}$/.test(window.LMV_VIEWER_VERSION) ? "isAvailable" : "getExtension"

        // If extension already registered
        if(AutodeskViewing.theExtensionManager[isExtensionAlreadyRegistered](name)){
            registeredExtensions.push(name);
            continue;
        }

        let ExtensionCtor = customExtensions[name];

        let extended = new ExtensionCtor(AutodeskViewing);
        
        let result = AutodeskViewing.theExtensionManager.registerExtension(name, extended);
        
        if (result === true)
            registeredExtensions.push(name);

    }

    return registeredExtensions;

}
```

Reply:
will merge this anyway, not as big of a project to support backwards compatibility jet 😂